File: mediumSpringWebflux/src/main/java/com/example/mediumspringwebflux/board/service/Facade/FindBoard.java
package com.example.mediumspringwebflux.board.service.Facade;

import com.example.mediumspringwebflux.board.document.Board;
import com.example.mediumspringwebflux.board.repository.BoardRepository;
import org.springframework.stereotype.Component;
import reactor.core.publisher.Mono;

@Component
public class FindBoard {
    private final BoardRepository boardRepository;

    public FindBoard(BoardRepository boardRepository) {
        this.boardRepository = boardRepository;
    }

    public Mono<Board> execute(String id) {
        return boardRepository.findById(id)
                .switchIfEmpty(Mono.error(new Exception("게시물을 찾지 못함")));
    }

}

File: mediumSpringWebfluxcorutine/src/main/kotlin/com/example/mediumspringwebfluxcorutine/board/service/facade/FindBoard.kt
package com.example.mediumspringwebfluxcorutine.board.service.facade

import com.example.mediumspringwebfluxcorutine.board.document.Board
import com.example.mediumspringwebfluxcorutine.board.repository.BoardRepository
import kotlinx.coroutines.reactor.awaitSingleOrNull
import org.springframework.stereotype.Component

@Component
class FindBoard(
    private val boardRepository: BoardRepository
) {

    suspend fun byId(id: String): Board {
        return boardRepository.findById(id).awaitSingleOrNull() ?: throw RuntimeException("게시물을 찾지 못함")
    }

}